modules/standard/GpuDiagnostics.chpl

```diff
@@ -56,6 +56,70 @@ module GpuDiagnostics
     var device_to_device: uint(64);
   };
 
+  /*
+   * Utility function to assert GPU diagnostic values. First 4 formals have the
+   * same names as the fields in chpl_gpuDiagnostics type. `_um` and `_aod`
+   * versions of `kernel_launch` are for convenience where we expect different
+   * number of launches in different memory modes
```

Comment:
It's not too hard to figure this out from looking at the code, but since most folks will just look at this top comment, you might want to add:

- -1 (and really any other negative value) signals that a given diagnostic should be ignored.
- if either 'kernel_launch_um' or 'kernel_launch_aod' are set then 'kernel_launch' launch will be ignored (though we do check for this and warn in such cases).
- For the time being, this is only considering gpu 0 on the current locale.

Reply:
> For the time being, this is only considering gpu 0 on the current locale

Actually, it accounts for all the GPUs in locale 0.

Reply:
Ah I see, that's right. The doc for gpuDiagnostics(): "array of counts of GPU ops initiated on each locale"

And we do `const diags = getGpuDiagnostics()[0];`

So I guess the thing to say is that for each diagnostic, we check the sum total across all GPUs on Locale 0.

Reply:
Yes. And arguably we might want to do it per-gpu. I imagine it is not a huge implementation lift, but a bit different than what we have today.